persistence/sql/persister_courier.go

```diff
@@ -25,6 +25,33 @@ func (p *Persister) AddMessage(ctx context.Context, m *courier.Message) error {
 	return sqlcon.HandleError(p.GetConnection(ctx).Create(m)) // do not create eager to avoid identity injection.
 }
 
+func (p *Persister) ListMessages(ctx context.Context, filter courier.MessagesFilter) ([]courier.Message, int64, error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.ListMessages")
+	defer span.End()
+
+	q := p.GetConnection(ctx).Where("nid=?", p.NetworkID(ctx))
+
+	if filter.Status != nil {
+		q = q.Where("status=?", *filter.Status)
+	}
+
+	if filter.Recipient != "" {
+		q = q.Where("recipient=?", filter.Recipient)
+	}
+
+	messages := make([]courier.Message, 0)
+	if err := q.Paginate(filter.Page, filter.PerPage).Order("created_at DESC").All(&messages); err != nil {
```

Comment:
I know this wasn't part of the issue, but seeing this line made me think of this: It might be worth adding a sort-direction to the `courier.MessagesFilter` struct to set the direction (`DESC` or `ASC`) here. This could be used to get the first sent emails matching the other filters. cc @aeneasr

Reply:
@jonas-jonas I think that's a good idea. Should i proceed to implement this now ? Or are we waiting for @aeneasr 's approval ?

Reply:
I agree that it would be useful, but since we do not have ordering yet in our REST APIs and we have no design document describing how such ordering would look like, i'd like to defer this for another time :)

Reply:
Then LGTM.